src/main/scala/net/snowflake/spark/snowflake/Utils.scala

```diff
@@ -724,4 +724,64 @@ object Utils {
     }
     // scalastyle:off println
   }
+
+  // Find the first un-quoted char '.', return -1 if not found
```

Comment:
Since you split this out as a separate function, should you have a couple of unit tests for it?